lib/internal/http2/compat.js

```diff
@@ -391,6 +391,8 @@ function onStreamCloseResponse() {
   state.closed = true;
 
   this[kProxySocket] = null;
+
+  this.off('wantTrailers', onStreamTrailersReady);
```

Comment:
Nit: would you be so kind and use `removeListener` instead? That seems much clearer from the semantics and is therefore easier to grasp :-)

Reply:
Done